process/transactions/transactions_processor_test.go

```diff
@@ -0,0 +1,325 @@
+		tx:         generateRandomTx(),
+		headerData: headerData,
+	}
+}
+
+func TestNewTransactionProcessor(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		args        func() (core.PubkeyConverter, hashing.Hasher, marshal.Marshalizer)
+		expectedErr error
+	}{
+		{
+			args: func() (core.PubkeyConverter, hashing.Hasher, marshal.Marshalizer) {
+				return nil, &mock.HasherStub{}, &mock.MarshallerStub{}
+			},
+			expectedErr: covalent.ErrNilPubKeyConverter,
+		},
+		{
+			args: func() (core.PubkeyConverter, hashing.Hasher, marshal.Marshalizer) {
+				return &mock.PubKeyConverterStub{}, nil, &mock.MarshallerStub{}
+			},
+			expectedErr: covalent.ErrNilHasher,
+		},
+		{
+			args: func() (core.PubkeyConverter, hashing.Hasher, marshal.Marshalizer) {
+				return &mock.PubKeyConverterStub{}, &mock.HasherStub{}, nil
+			},
+			expectedErr: covalent.ErrNilMarshaller,
+		},
+		{
+			args: func() (core.PubkeyConverter, hashing.Hasher, marshal.Marshalizer) {
+				return &mock.PubKeyConverterStub{}, &mock.HasherStub{}, &mock.MarshallerStub{}
+			},
+			expectedErr: nil,
+		},
+	}
+
+	for _, currTest := range tests {
+		_, err := transactions.NewTransactionProcessor(currTest.args())
+		require.Equal(t, currTest.expectedErr, err)
+	}
+}
+
+func TestTransactionProcessor_ProcessTransactions_InvalidBody_ExpectError(t *testing.T) {
+	hData := generateRandomHeaderData()
+	txPool := map[string]data.TransactionHandler{}
+	body := data.BodyHandler(nil)
+
+	txp, _ := transactions.NewTransactionProcessor(&mock.PubKeyConverterStub{}, &mock.HasherStub{}, &mock.MarshallerStub{})
+	_, err := txp.ProcessTransactions(hData.header, hData.headerHash, body, txPool)
+
+	require.Equal(t, err, covalent.ErrBlockBodyAssertion)
+}
+
+func TestTransactionProcessor_ProcessTransactions_InvalidMarshaller_ExpectError(t *testing.T) {
+	hData := generateRandomHeaderData()
+	txPool := map[string]data.TransactionHandler{}
+	body := &block.Body{MiniBlocks: []*block.MiniBlock{{Type: block.TxBlock}}}
+
+	errMarshaller := errors.New("err marshaller")
+	txp, _ := transactions.NewTransactionProcessor(
+		&mock.PubKeyConverterStub{},
+		&mock.HasherStub{},
+		&mock.MarshallerStub{
+			MarshalCalled: func(obj interface{}) ([]byte, error) {
+				return nil, errMarshaller
+			},
+		})
+	_, err := txp.ProcessTransactions(hData.header, hData.headerHash, body, txPool)
+
+	require.Equal(t, err, errMarshaller)
+}
+
+func TestTransactionProcessor_ProcessTransactions_OneEmptyTxBlock_ExpectZeroProcessedTxs(t *testing.T) {
+	hData := generateRandomHeaderData()
+
+	body := &block.Body{MiniBlocks: []*block.MiniBlock{
+		{
+			TxHashes:        [][]byte{},
+			ReceiverShardID: 1,
+			SenderShardID:   2,
+			Type:            block.TxBlock},
+	},
+	}
+
+	txp, _ := transactions.NewTransactionProcessor(&mock.PubKeyConverterStub{}, &mock.HasherStub{}, &mock.MarshallerStub{})
+	ret, _ := txp.ProcessTransactions(hData.header, hData.headerHash, body, map[string]data.TransactionHandler{})
```

Comment:
can you check also err?
it might be 0 len because of returned err

Reply:
Yes, changed it everywhere to also check for error